_posts/2018-08-20-how-we-use-pivotal-tracker-at-ombu-labs.markdown

```diff
@@ -0,0 +1,144 @@
+---
+layout: post
+title:  "How we use Pivotal Tracker at Ombu Labs"
+date: 2018-08-20 11:10:00
+categories: ["agile", "pivotal-tracker"]
+author: "etagwerker"
+---
+
+We tend to use [GitHub](https://github.com) for everything at [Ombu Labs](https://www.ombulabs.com), but sometimes our clients
+prefer [Pivotal Tracker](https://www.pivotaltracker.com). We have no problem
+with Pivotal Tracker, we just like to keep everything in one place, that is why
+we prefer GitHub for everything.
+
+There are many things that are easier with Pivotal Tracker, as long as you are
+using it the right way. Some of its features are very useful for agile teams like
+ours. This is how we like to use it to ship value with every sprint and keep
+track of our [velocity](http://wiki.c2.com/?ProjectVelocity).
+
+<!--more-->
+
+## Step 1: Filing user stories
+
+Every thing has to be tracked according to the type of story. Most of the user
+stories are either *features* or *bugs*, although sometimes there will be *chores* to be
+done. If a *chore* reoccurs many times, then it's probably time to write a *feature*
+to reduce the time spent completing that *chore*.
+
+## Step 2: Describing user stories
+
+Every [user story](https://www.mountaingoatsoftware.com/agile/user-stories) should
+be a simple description of a feature:
+
+```
+As a < type of user >, I want < some goal > so that < some business goal >.
+```
+
+The description of the user story should be very clear about the expected outcome.
+This will explain what is the exit criteria for the story. It is very helpful
+for designers and developers to understand what needs to happen for a user to
+be happy with the feature.
+
+This is an example from one of our recent projects ([https://audit.fastruby.io/](https://audit.fastruby.io/)):
+
+```
+As a user I want to submit my Gemfile.lock
+so that I can see an audit analysis of its vulnerabilities.
+```
+
+<img src="/blog/assets/images/user-story-audit-fastruby-io.png" alt="Example of a simple user story" class="medium-img">
+
+If the client requested this feature, you should add them as the *Requester* of
+the story. If it was you, then you can add yourself to it.
+
+## Step 3: Estimating user stories
+
+Before you start working on a user story, you will need to estimate it. You
+can estimate it with 0, 1, 2, 3, 5, or 8 points. We prefer to use a [Fibonacci](https://en.wikipedia.org/wiki/Fibonacci_number)
+sequence to estimate stories.
+
+If a story is more complicated than 8 points, you should break it down in two or
+more stories. When you estimate, you must remember that you are [estimating
+complexity](https://rubygarage.org/blog/3-reasons-to-estimate-with-story-points),
+**not hours**. Different developers might take different amount of time
+in completing that story, so **we shouldn't estimate hours**.
+
+For the example we mentioned before, we estimate that that story has 3 points of
+complexity.
```

Comment:
I think this is something we've discussed in the past, but shouldn't there be a points to hours mapping? Or does that come after the team has worked on one or more sprints, and you can tell from the velocity and # of hours worked? 

I get the idea of using points to estimate features/bugs, but in certain scenarios (when we estimate Rails Upgrades or MVPs) we use hours, not points.

Reply:
We should always use points. I'll add a paragraph to explain a bit more about that.

I believe points give us more flexibility in terms of calculating when a story will be done.